types/storybook-readme/index.d.ts

```diff
@@ -16,10 +16,12 @@ export type DecoratorPattern = (
 
 export type HOCPattern = (story: RenderFunction) => Renderable | null;
 
+export type MakeDecoratorResult = (...args: any) => any;
+
 // Types added for v5
-export type addReadme = (decorator: DecoratorPattern) => React.ReactNode;
-export type addFooter = (md: string) => void;
-export type addHeader = (md: string) => void;
+export const addReadme: MakeDecoratorResult
```

Comment:
I'm not really sure If the type should be like this.  On storybook-readme code,  `const addReadme = makeDecorator({ ... }) `  So It is the return value of `makeDecorater` and It is from https://github.com/storybooks/storybook/blob/next/lib/addons/src/make-decorator.ts#L38   with the type of return value is `type MakeDecoratorResult = (...args: any) => any;`

Reply:
That seems like the best you can do for now. I would suggest asking storybook to export `MakeDecoratorResult`, so you could import that here and stay in sync with future changes.